### PR TITLE
allow to hide email addresses in the sharing search dialog

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
mask emails by default

## Related Issue

- uses https://github.com/owncloud/ocis/pull/8781 and https://github.com/owncloud/ocis/pull/8825/commits/12c785e5846e66ec4643f5f1e0b4d942c0519427  from https://github.com/owncloud/ocis/pull/8825
- needs https://github.com/owncloud/ocis-charts/pull/524 first for chart and application version bump

## How Has This Been Tested?
- use the development example with this modification:

```diff
diff --git a/deployments/development-install/helmfile.yaml b/deployments/development-install/helmfile.yaml
index 762c62d..a38d76c 100644
--- a/deployments/development-install/helmfile.yaml
+++ b/deployments/development-install/helmfile.yaml
@@ -25,6 +25,16 @@ releases:
           demoUsers: true
           basicAuthentication: true
 
+      - image:
+          repository: wkloucek/ocis
+          tag: release-5.0.1-8d1ba0244c3203c9bd2ac05b683d01afe885fc8d
+
+      - features:
+          sharing:
+            users:
+              search:
+                showUserEmail: false # you can also use true = the chart default to show them
+
       - services:
           idm:
             persistence:
```

then create another user and share a file with this user. In the sharing dialog, you should see 

- if showUserEmail is set to true:

image

- if showUserEmail is set to false:

image

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation generated (`make docs`) and committed
- [ ] Documentation ticket raised: <link>
- [ ] Documentation PR created: <link>

### CODE REVIEW

**Comment by wkloucek** [comment]
> Hmm, need to wait for feedback on https://github.com/owncloud/ocis/pull/8781/files#r1559455915

fix in https://github.com/owncloud/ocis/pull/8825/commits/12c785e5846e66ec4643f5f1e0b4d942c0519427  from https://github.com/owncloud/ocis/pull/8825

**Comment by wkloucek** [comment]
@d7oc this needs a release of oCIS 5.0.1 and merge of #524 first. But from my side it's ready for review (I'll keep it in a draft state until mergeable though)

File: charts/ocis/templates/frontend/deployment.yaml [hunk 1]
```diff
@@ -98,9 +98,13 @@ spec:
             - name: FRONTEND_FULL_TEXT_SEARCH_ENABLED
               value: {{ not (eq .Values.services.search.extractor.type "basic") | quote }}
 
+            - name: FRONTEND_SHOW_USER_EMAIL_IN_RESULTS
+              value: {{ .Values.features.sharing.users.search.showUserEmail | quote }}
+
             - name: OCIS_DISABLE_SSE
               value: {{ .Values.features.sse.disabled | quote }}
 
+
```
Comment (d7oc) [inline charts/ocis/templates/frontend/deployment.yaml:107]: Why to we have this additional whitespace here?

**Review by d7oc — COMMENTED** [review]
LGTM aside formatting.

File: charts/ocis/templates/frontend/deployment.yaml (hunk 1 above)
Comment (wkloucek) [reply charts/ocis/templates/frontend/deployment.yaml]: fixed https://github.com/owncloud/ocis-charts/pull/526/commits/a771ac9debf6b5b8c495fb0b5bed686021206da7

**Review by d7oc — APPROVED** [review]
LGTM

**Comment by wkloucek** [comment]
tested again after merging main with #524 -> works like a charm

**Pull request merged**